src/main/resources/data/computercraft/lua/rom/apis/rednet.lua

```diff
@@ -381,13 +395,14 @@ function run()
         if sEvent == "modem_message" then
             -- Got a modem message, process it and add it to the rednet event queue
             local sModem, nChannel, nReplyChannel, tMessage = p1, p2, p3, p4
-            if isOpen(sModem) and (nChannel == os.getComputerID() or nChannel == CHANNEL_BROADCAST) then
+            if isOpen(sModem) and (nChannel == idAsChannel() or nChannel == CHANNEL_BROADCAST) then
                 if type(tMessage) == "table" and type(tMessage.nMessageID) == "number"
                     and tMessage.nMessageID == tMessage.nMessageID and not tReceivedMessages[tMessage.nMessageID]
+                    and tMessage.nRecipient and tMessage.nRecipient == os.getComputerID() or nChannel == CHANNEL_BROADCAST
```

Comment:
Nitpick I know, but maybe

```suggestion
                    and ((tMessage.nRecipient and tMessage.nRecipient == os.getComputerID()) or nChannel == CHANNEL_BROADCAST)
```

Could also probably drop the the `tMessage.nRecipient and `, but maybe not worth it. Sorry, I just always forget `and`/`or` precedence so like to keep these things explicit.

Reply:
This change breaks linter.